select_assignment2_flicks/Model/Movie.swift

```diff
@@ -41,6 +42,39 @@ struct Movie {
     
     var voteAverage: Double = 0
     
+    var myRootRef = Firebase(url: Constants.FIREBASE_URL)
+    
+    func isFavoriteFireBase(onComplete: (list: [String], value: Bool) -> Void) {
+        let deviceRef = myRootRef.childByAppendingPath(UIDevice.currentDevice().identifierForVendor!.UUIDString)
+        
+        let favoritesRef = deviceRef.childByAppendingPath("favorites")
+        
+        favoritesRef.observeEventType(.Value, withBlock: { snapshot in
```

Comment:
the way this is written, this function will be called many times. so this should be `observeSingleEventOfType`